### PR TITLE
Tweak dvc pr

### LINKED ISSUE #41: More git integration
@gabrieltorresgamez @yvokeller 

- [x] [https://cml.dev/](https://cml.dev/)
- [x] Github Pipeline (self hosted?)
- [x] self-hosted iterative.ai instance, was das sinn? welche vorteile haben wir?

### PR DESCRIPTION
**Pull request by gabrieltorresgamez**
add the step, where a pull request is created after an automatized run

### CODE REVIEW

**Comment by ghost** [comment]
# RUN: 2023-04-21 23-17-04

## Hyperparameters

| Path   | Param   | HEAD^   | workspace   |
|--------|---------|---------|-------------|

## Metrics

| Path   | Metric   | HEAD^   | workspace   | Change   |
|--------|----------|---------|-------------|----------|

# Plots

TODO

**Comment by ghost** [comment]
# RUN: 2023-04-21 23-36-50

## Hyperparameters

| Path        | Param                               | HEAD^   | workspace   |
|-------------|-------------------------------------|---------|-------------|
| params.yaml | dvclive.model_hparams.learning_rate | 0.03    | 0.1         |
| params.yaml | dvclive.model_hparams.weight_decay  | 0.001   | 0.0005      |

## Metrics

| Path                 | Metric           | HEAD^   | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | test.epoch.acc   | 0.44875 | 0.3491      | -0.09964 |
| dvclive/metrics.json | test.epoch.f1    | 0.36818 | 0.25647     | -0.1117  |
| dvclive/metrics.json | test.step.acc    | 0.44875 | 0.3491      | -0.09964 |
| dvclive/metrics.json | test.step.f1     | 0.36818 | 0.25647     | -0.1117  |
| dvclive/metrics.json | train.epoch.acc  | 0.73694 | 0.81162     | 0.07468  |
| dvclive/metrics.json | train.epoch.f1   | 0.65581 | 0.76306     | 0.10725  |
| dvclive/metrics.json | train.epoch.loss | 0.74477 | 0.53597     | -0.2088  |
| dvclive/metrics.json | train.step.f1    | 0.89444 | 0.93225     | 0.03781  |
| dvclive/metrics.json | train.step.loss  | 0.29613 | 0.30755     | 0.01142  |
| dvclive/metrics.json | val.epoch.acc    | 0.82049 | 0.85795     | 0.03746  |
| dvclive/metrics.json | val.epoch.f1     | 0.79137 | 0.83348     | 0.04211  |
| dvclive/metrics.json | val.epoch.loss   | 0.55358 | 0.43421     | -0.11937 |
| dvclive/metrics.json | val.step.f1      | 0.66667 | 0.64286     | -0.02381 |
| dvclive/metrics.json | val.step.loss    | 0.95916 | 0.23401     | -0.72515 |

# Plots

TODO

**Comment by ghost** [comment]
# RUN: 2023-04-21 23-44-30

## Hyperparameters

| Path        | Param                            | HEAD^   | workspace   |
|-------------|----------------------------------|---------|-------------|
| params.yaml | dvclive.model_hparams.num_epochs | 2       | 3           |

## Metrics

| Path                 | Metric           | HEAD^   | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | epoch            | 2       | 3           | 1        |
| dvclive/metrics.json | step             | 354     | 531         | 177      |
| dvclive/metrics.json | test.epoch.acc   | 0.3491  | 0.37706     | 0.02796  |
| dvclive/metrics.json | test.epoch.f1    | 0.25647 | 0.29407     | 0.03759  |
| dvclive/metrics.json | test.step.acc    | 0.3491  | 0.37706     | 0.02796  |
| dvclive/metrics.json | test.step.f1     | 0.25647 | 0.29407     | 0.03759  |
| dvclive/metrics.json | train.epoch.acc  | 0.81162 | 0.89319     | 0.08157  |
| dvclive/metrics.json | train.epoch.f1   | 0.76306 | 0.85782     | 0.09476  |
| dvclive/metrics.json | train.epoch.loss | 0.53597 | 0.33408     | -0.20189 |
| dvclive/metrics.json | train.step.acc   | 0.9375  | 0.96875     | 0.03125  |
| dvclive/metrics.json | train.step.f1    | 0.93225 | 0.95789     | 0.02564  |
| dvclive/metrics.json | train.step.loss  | 0.30755 | 0.10856     | -0.19899 |
| dvclive/metrics.json | val.epoch.acc    | 0.85795 | 0.89823     | 0.04028  |
| dvclive/metrics.json | val.epoch.f1     | 0.83348 | 0.87489     | 0.04141  |
| dvclive/metrics.json | val.epoch.loss   | 0.43421 | 0.32432     | -0.10989 |
| dvclive/metrics.json | val.step.f1      | 0.64286 | 0.66667     | 0.02381  |
| dvclive/metrics.json | val.step.loss    | 0.23401 | 0.78875     | 0.55474  |

# Plots

TODO

**Comment by ghost** [comment]
# RUN: 2023-04-21 23-57-17

## Hyperparameters

| Path        | Param                            | HEAD^   | workspace   |
|-------------|----------------------------------|---------|-------------|
| params.yaml | dvclive.data.batch_size          | 32      | 64          |
| params.yaml | dvclive.model_hparams.num_epochs | 3       | 10          |

## Metrics

| Path                 | Metric           | HEAD^   | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | epoch            | 3       | 10          | 7        |
| dvclive/metrics.json | step             | 531     | 890         | 359      |
| dvclive/metrics.json | test.epoch.acc   | 0.37706 | 0.38351     | 0.00645  |
| dvclive/metrics.json | test.epoch.f1    | 0.29407 | 0.31615     | 0.02209  |
| dvclive/metrics.json | test.step.acc    | 0.37706 | 0.38351     | 0.00645  |
| dvclive/metrics.json | test.step.f1     | 0.29407 | 0.31615     | 0.02209  |
| dvclive/metrics.json | train.epoch.acc  | 0.89319 | 0.96381     | 0.07062  |
| dvclive/metrics.json | train.epoch.f1   | 0.85782 | 0.9572      | 0.09938  |
| dvclive/metrics.json | train.epoch.loss | 0.33408 | 0.10953     | -0.22455 |
| dvclive/metrics.json | train.step.acc   | 0.96875 | 1.0         | 0.03125  |
| dvclive/metrics.json | train.step.f1    | 0.95789 | 1.0         | 0.04211  |
| dvclive/metrics.json | train.step.loss  | 0.10856 | 0.01936     | -0.0892  |
| dvclive/metrics.json | val.epoch.acc    | 0.89823 | 0.91378     | 0.01555  |
| dvclive/metrics.json | val.epoch.f1     | 0.87489 | 0.906       | 0.03111  |
| dvclive/metrics.json | val.epoch.loss   | 0.32432 | 0.33504     | 0.01072  |
| dvclive/metrics.json | val.step.loss    | 0.78875 | 0.39189     | -0.39686 |

# Plots

TODO

**Comment by gabrieltorresgamez** [comment]
soooooo big men

Ab jetzt wird jedes mal, wenn sich die Hyperparameter und/oder Scores(bei Codeänderungen) verändern, ein neuer Pull Request gemacht. Dieser kann danach bei besserem Score gemerged oder abgelehnt werden.

Falls man jetzt aber normale Codeänderungen gemacht hat, welche das Modell nicht beinflussen, wir kein Pull Request und auch kein leerer Kommentar generiert.

Nach dem mergen kann man im Iterative Studio ganz einfach die Metriken zweier Modelle vergleichen und deren Verlauf der Loss/Acc/F1 function visualisiert werden.

thats all i think, das nutzen von Studio sollte jetzt problemlos(er) sein :)
Bei Problemen gerne nachfragen

**Review by florinbarbisch — APPROVED** [review]

**Comment by ghost** [comment]
# RUN: 2023-04-23 12-49-29

## Hyperparameters

| Path        | Param                            | HEAD^   | workspace   |
|-------------|----------------------------------|---------|-------------|
| params.yaml | dvclive.data.batch_size          | 32      | 64          |
| params.yaml | dvclive.model_hparams.num_epochs | 2       | 10          |

## Metrics

| Path                 | Metric           | HEAD^   | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | epoch            | 2       | 10          | 8        |
| dvclive/metrics.json | step             | 354     | 890         | 536      |
| dvclive/metrics.json | test.epoch.acc   | 0.36129 | 0.38351     | 0.02222  |
| dvclive/metrics.json | test.epoch.f1    | 0.26923 | 0.31615     | 0.04692  |
| dvclive/metrics.json | test.step.acc    | 0.36129 | 0.38351     | 0.02222  |
| dvclive/metrics.json | test.step.f1     | 0.26923 | 0.31615     | 0.04692  |
| dvclive/metrics.json | train.epoch.acc  | 0.82433 | 0.96381     | 0.13948  |
| dvclive/metrics.json | train.epoch.f1   | 0.76904 | 0.9572      | 0.18816  |
| dvclive/metrics.json | train.epoch.loss | 0.50873 | 0.10953     | -0.3992  |
| dvclive/metrics.json | train.step.acc   | 0.96875 | 1.0         | 0.03125  |
| dvclive/metrics.json | train.step.f1    | 0.95789 | 1.0         | 0.04211  |
| dvclive/metrics.json | train.step.loss  | 0.15193 | 0.01936     | -0.13258 |
| dvclive/metrics.json | val.epoch.acc    | 0.8311  | 0.91378     | 0.08269  |
| dvclive/metrics.json | val.epoch.f1     | 0.7995  | 0.906       | 0.10651  |
| dvclive/metrics.json | val.epoch.loss   | 0.62941 | 0.33504     | -0.29436 |
| dvclive/metrics.json | val.step.acc     | 0.71429 | 0.85714     | 0.14286  |
| dvclive/metrics.json | val.step.f1      | 0.5     | 0.66667     | 0.16667  |
| dvclive/metrics.json | val.step.loss    | 0.98827 | 0.39189     | -0.59638 |

# Plots

TODO
